Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/workspace/ExternalWSManager.java

```diff
@@ -0,0 +1,89 @@
+package edu.asu.spring.quadriga.service.impl.workspace;
+
+import java.util.Date;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import edu.asu.spring.quadriga.dao.IUserDAO;
+import edu.asu.spring.quadriga.dao.workbench.IProjectDAO;
+import edu.asu.spring.quadriga.dao.workspace.IExternalWsDAO;
+import edu.asu.spring.quadriga.domain.IUser;
+import edu.asu.spring.quadriga.dto.ExternalWorkspaceDTO;
+import edu.asu.spring.quadriga.dto.ProjectDTO;
+import edu.asu.spring.quadriga.dto.ProjectWorkspaceDTO;
+import edu.asu.spring.quadriga.dto.QuadrigaUserDTO;
+import edu.asu.spring.quadriga.exceptions.QuadrigaAccessException;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.mapper.ProjectDTOMapper;
+import edu.asu.spring.quadriga.passthroughproject.constants.Constants;
+import edu.asu.spring.quadriga.service.workspace.IExternalWSManager;
+
+/**
+ * 
+ * This class has all the service layer functions to modify and get external
+ * workspace information.
+ * 
+ *
+ */
+@Service
+public class ExternalWSManager implements IExternalWSManager {
+
+    @Autowired
+    private IExternalWsDAO externalWorkspaceDAO;
+
+    @Autowired
+    private IProjectDAO projectDao;
+
+    @Autowired
+    private ProjectDTOMapper projectMapper;
+
+    @Autowired
+    private IUserDAO userDAO;
+
+    @Override
+    public boolean isExternalWorkspaceExists(String externalWorkspaceId)
+            throws QuadrigaStorageException, QuadrigaAccessException {
+        ExternalWorkspaceDTO externalWorkspaceDTO = externalWorkspaceDAO.getExternalWorkspace(externalWorkspaceId);
+        if (externalWorkspaceDTO != null) {
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public void createExternalWorkspace(String externalWorkspaceId, String externalWorkspaceName, String workspaceId,
```

Comment:
why does this method accept an id for a workspace? the manager should create the workspace id.